### PR TITLE
Strict deps java cmd line flag

### PR DESCRIPTION
**Pull request by ittaiz**
strict_java_deps for java can be configured via command line
This was pending a [change](https://github.com/bazelbuild/bazel/commit/cd1ae23c72a206461c4c8b431dd2484727fb2afc) in Bazel to be released in 0.6.0

### CODE REVIEW

**Comment by ittaiz** [comment]
@or-shachar wdyt?

File: test_run.sh [hunk 1]
```diff
@@ -583,3 +592,4 @@ $runner test_scala_library_expect_no_recompilation_on_internal_change_of_java_de
 $runner test_scala_library_expect_no_java_recompilation_on_internal_change_of_scala_sibling
 $runner test_scala_library_expect_failure_on_missing_direct_java
 $runner bazel run test:test_scala_proto_server
+$runner test_scala_library_expect_failure_on_missing_direct_deps_warn_mode_java
```
[the remark this answers is not in the document]
Comment (ittaiz) [reply test_run.sh]: done

**Comment by johnynek** [comment]
@ittaiz when do you think we should add some docs on using strict deps? I’d like to benchmark it on our largest repo and see if what the perf cost is and what it finds.

**Comment by ittaiz** [comment]
see #329

**Pull request merged**